python/lsst/ts/dsm/dsm_csc.py

```diff
@@ -1,34 +1,388 @@
+                    self.log.debug("Done row.")
+                except IndexError as error:
+                    self.log.error(f"{ifile}: {error}")
+
+    def process_event(self, event):
+        """Process I/O Events.
+
+        Parameters
+        ----------
+        event : `aionotify.Event`
+            Payload containing the I/O event information.
+        """
+        self.log.debug(f"Event: Flags = {event.flags}, Name = {event.name}")
+        if event.flags & aionotify.Flags.CLOSE_WRITE:
+            if event.name.endswith("yaml"):
+                self.process_yaml_file(event.name)
+            if event.name.endswith("dat"):
+                self.process_dat_file(event.name)
+
+    def process_yaml_file(self, ifile):
+        """Process the UI configuration YAML file and send telemetry.
+
+        Parameters
+        ----------
+        ifile : `str`
+          The filename to read and process.
+        """
+        self.log.info(f"Process {ifile} file.")
+        with open(os.path.join(self.telemetry_directory, ifile), 'r') as infile:
+            self.log.debug("UI Config file opened.")
+            content = yaml.safe_load(infile)
+            ui_config_file = pathlib.PosixPath(content['ui_versions']['config_file']).as_uri()
+            self.tel_configuration.set_put(dsmIndex=self.salinfo.index,
+                                           timestampConfigStart=utils.convert_time(content['timestamp']),
+                                           uiVersionCode=content['ui_versions']['code'],
+                                           uiVersionConfig=content['ui_versions']['config'],
+                                           uiConfigFile=ui_config_file,
+                                           cameraName=content['camera']['name'],
+                                           cameraFps=content['camera']['fps'],
+                                           dataBufferSize=content['data']['buffer_size'],
+                                           dataAcquisitionTime=content['data']['acquisition_time'])
+            self.log.info("Done with UI config file.")
+
+    def report_summary_state(self):
+        """Handle things that depend on state.
+        """
+        self.log.info(f"Current state: {self.summary_state}")
+        if self.summary_state in (salobj.State.DISABLED, salobj.State.ENABLED):
+            if self.telemetry_loop_task.done():
+                self.log.info("Adding telemetry loop task.")
+                self.telemetry_loop_task = asyncio.ensure_future(self.telemetry_loop())
+
+            if self.simulation_mode and self.simulated_telemetry_loop_task.done():
+                self.log.info("Adding simulated telemetry loop task.")
+                self.simulated_telemetry_loop_task = asyncio.ensure_future(self.simulated_telemetry_loop())
+        else:
+            if self.simulation_mode and self.telemetry_directory is None:
+                self.telemetry_directory = tempfile.mkdtemp()
```

Comment:
Consider creating `telemetry_directory` in `implement_summary_state` (if simulating, no need to test the current state). That way it'll be created as soon as it can possibly be needed.

Reply:
Is `implement_summary_state` in salobj v5? Or did you mean do this in `implement_simulation_mode`?

Reply:
I meant the latter